lib/jcsda-emc/spack-stack/stack/container_env.py

```diff
@@ -48,16 +48,18 @@ def write(self):
             # but it's not understood when used in a spack.yaml
             filedata = f.read()
             filedata.replace('::', ':')
+            filedata = filedata.replace('::', ':')
             app_yaml = syaml.load_config(filedata)
 
         with open(self.container_path, 'r') as f:
             container_yaml = syaml.load_config(f)
 
+        # Create copy so we can modify it
         original_yaml = copy.deepcopy(container_yaml)
 
         with open(self.base_packages, 'r') as f:
             filedata = f.read()
-            filedata.replace('::', ':')
+            filedata = filedata.replace('::', ':')
```

Comment:
We may need this for environments as well ... will check again with your latest code in a bit.

Reply:
If using includes it should not be necessary.

Reply:
True for preconfigured sites, but for others when we need to run `spack config add "packages packages:all:providers:mpi:[openmpi@4.1.3]" and similar for compilers, we can't use double colons. I've seen in the past that on macOS the entry `packages:all:compiler:[gcc]` is ignored and `clang` is used, unless double colons are added.

Reply:
So, I have some code that adds this. Are you saying instead of putting it in the main `spack.yaml` I should insert it as a double colon value in `site/packages.yaml` instead?

Reply:
I believe this would work. But let's try first with your code as-is, we can fix that in a follow-up PR if needed?

Reply:
Sure, it should be fine as is then.

Reply:
I'll take another look at the container build and if that one is ok then we can merge!

Reply:
The `app` label is unfortunately still empty, and the order of the package includes reversed (but that doesn't matter). I will rearrange the comments in the container yaml in my next PR so that they make more sense, because now we have:
```
  # Basic package config from configs/common/packages.yaml
  # Additional package config for container
  packages:
    gcc:
      buildable: false
      externals:
      - spec: gcc@10
        prefix: /usr
    git:
      buildable: false
      externals:
      - spec: git@2
        prefix: /usr
    git-lfs:
      buildable: false
      externals:
      - spec: git-lfs@2
        prefix: /usr
... all other packages
```
used to be:
```
  # Basic package config from configs/common/packages.yaml
  ... all other packages
  # Additional package config for container
 ...
```

Not sure what to do about `app`, though.

Reply:
Fixed the app label.

Reply:
Could put an empty `packages: {}` and then when the yaml is merged it'll remain in the same location.

Reply:
> Fixed the app label.

Yay! Works.